src/settings/LocationLocations/LocationManager.js

```diff
@@ -459,58 +460,50 @@ class LocationManager extends React.Component {
     this.transitionToParams({ layer: 'clone' });
   };
 
+  checkLocationHasHoldingsOrItems = async (locationId) => {
+    const { mutator } = this.props;
+    const query = `permanentLocationId=${locationId} or temporaryLocationId=${locationId}`;
+
+    mutator.holdingsEntries.reset();
+    mutator.itemEntries.reset();
+
+    const results = await Promise.all([
+      mutator.holdingsEntries.GET({ params: { query } }),
+      mutator.itemEntries.GET({ params: { query } }),
+    ]);
+
+    return results.some(records => records.length > 0);
+  };
+
   onRemove = location => {
     const {
       match,
       mutator,
     } = this.props;
-    const promises = [];
-
-    // Uses in Inventory
-    {
-      mutator.holdingsEntries.reset();
-      mutator.itemEntries.reset();
-      const query = `permanentLocationId=${location.id} or temporaryLocationId=${location.id}`;
-      promises.push(mutator.holdingsEntries.GET({ params: { query } }));
-      promises.push(mutator.itemEntries.GET({ params: { query } }));
-    }
 
-    return Promise.all(promises)
-      .then(values => {
-        if (undefined === values.find(records => records.length !== 0)) {
-          return mutator.entries.DELETE(location);
-        }
+    return this.checkLocationHasHoldingsOrItems(location.id)
+      .then(itHas => !itHas && mutator.entries.DELETE(location))
+      .then(result => {
+        const isRemoved = (result !== false);
 
-        return Promise.resolve(false);
-      }).then((isRemoved) => {
-        if (isRemoved !== false) {
+        if (isRemoved) {
           this.showCalloutMessage(location.name);
           this.transitionToParams({
             _path: `${match.path}`,
             layer: null
           });
-
-          return Promise.resolve(true);
         }
 
-        return Promise.resolve(false);
+        return isRemoved;
       });
   };
 
-  onSave = location => {
-    const { match } = this.props;
-
-    const action = location.id ? 'PUT' : 'POST';
-
-    return this.props.mutator.entries[action](location)
-      .then(updatedLocation => {
-        this.transitionToParams({
-          _path: `${match.path}/${updatedLocation.id}`,
-          layer: null,
-        });
-        this.setState({ selectedId: updatedLocation.id });
-      })
-      .catch(error => this.showSubmitErrorCallout(error.message || error.statusText));
+  updateSelected = location => {
+    this.transitionToParams({
+      _path: `${this.props.match.path}/${location.id}`,
+      layer: null,
+    });
+    this.setState({ selectedId: location.id });
```

Comment:
doesn't it cause problem in case route is changed at line above and then you change the state?